Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+        GL:warning("Nothing to broadcast, import Plus Ones data first!");
+        return false;
+    end
+
+    -- No need to keep any queued updates, we're doing a full broadcast now anyways
+    self.QueuedUpdates = {};
+
+    self.broadcastInProgress = true;
+    GL.Events:fire("GL.PLUSONES_BROADCAST_STARTED");
+
+    local Broadcast = function ()
+        GL:message("Broadcasting PlusOnes data...");
+
+        local Label = GL.Interface:get(GL.PlusOnes, "Label.BroadcastProgress");
+
+        if (Label) then
+            Label:SetText("Broadcasting...");
+        end
+
+        GL.CommMessage.new(
+            CommActions.broadcastPlusOnesData,
+            {
+                importString = self:export(false),
+                MetaData = DB:get("PlusOnes.MetaData", {}),
+            },
+            "GROUP"
+        ):send(function ()
+            GL:success("PlusOnes broadcast finished");
+
+            --- Broadcast updates before we reset the flag. Enabling this ends up calling receiveUpdate before receiveBroadcast and breaks the code
+           -- self:broadcastQueuedUpdates();
+
+            self.broadcastInProgress = false;
+            GL.Events:fire("GL.PLUSONES_BROADCAST_ENDED");
+
+            Label = GL.Interface:get(GL.PlusOnes, "Label.BroadcastProgress");
+            if (Label) then
+                Label:SetText("Broadcast finished!");
+            end
+        end, function (sent, plusOne)
+            Label = GL.Interface:get(GL.PlusOnes, "Label.BroadcastProgress");
+            if (Label) then
+                Label:SetText(string.format("Sent %s of %s bytes", sent, plusOne));
+            end
+        end);
+    end
+
+    -- We're about to send a lot of data which will put strain on CTL
+    -- Make sure we're out of combat before doing so!
+    if (UnitAffectingCombat("player")) then
+        GL:message("You are currently in combat, delaying PlusOnes broadcast");
+
+        GL.Events:register("PlusOnesOutOfCombatListener", "PLAYER_REGEN_ENABLED", function ()
+            GL.Events:unregister("PlusOnesOutOfCombatListener");
+            Broadcast();
+        end);
+    else
+        Broadcast();
+    end
+
+    return true;
+end
+
+--- Process an incoming plus one broadcast
+---
+---@param CommMessage CommMessage
+function PlusOnes:receiveBroadcast(CommMessage)
+    GL:debug("PlusOnes:receiveBroadcast");
+
+    -- No need to update our tables if we broadcasted them ourselves
+    if (CommMessage.Sender.name == GL.User.name) then
+        GL:debug("PlusOnes:receiveBroadcast received by self, skip");
+        return true;
+    end
+    local importString = CommMessage.content.importString or '';
+    local MetaData = CommMessage.content.MetaData or {};
+    local importBroadcast = (function ()
+        if (GL:empty(importString)) then
+            self:clearPlusOnes();
+            return;
+            --GL:warning("Couldn't process PlusOnes data received from " .. CommMessage.Sender.name);
```

Comment:
What happens if the broadcast is empty?

Reply:
What happens in the case of Plus Ones, is it allows clearing of all data when the ML or assist clears the plus one data.  For example, if I do multiple raids in one night and I want to clear the plus ones between the raids (i.e. do EoE then head to OS).  The commented section was the old code for rejecting the broadcast if the boosted roll data was empty.  I had done a search and replace of BoostedRolls with PlusOnes